File: assignor/src/main/java/com/uis/assignor/AssignorDecouple.java
package com.uis.assignor;

import com.uis.assignor.couple.Couple;
import com.uis.assignor.couple.SimpleDecouple;
import com.uis.decouple.BindCouple;

@BindCouple("Assignor")
public class AssignorDecouple extends SimpleDecouple {
    @Override
    public void onDecouple(Couple.Params param) {
        Couple.newResult(param.id).success().addParam("name","assignor onDecouple").notifyResult();
    }

    @Override
    public Couple.Result onCall(Couple.Params param) {
        return Couple.newResult(param.id).success().addParam("name","assignor onCall").build();
    }
}

File: demo/src/main/java/com/uis/conn/AssignorDecouple.kt
/*
 * Copyright (c) 2020 by uis
 * Author: uis
 * Github: https://github.com/luiing
 */

package com.uis.conn

import com.uis.assignor.call.Call
import com.uis.assignor.call.SimpleCall
import com.uis.decouple.BindCouple

@BindCouple("Assignor")
class AssignorDecouple : SimpleCall() {
    override fun onCallback(param: Call.Params) {
        Call.newResult(param.id).success().build()
    }

    override fun onCall(param: Call.Params): Call.Result {
        return Call.newResult(param.id).success().build()
    }
}
